### ISSUE TITLE
Merge models together

### ISSUE THREAD

**Issue by Shlomke** [fchollet/keras#3497]
Hi there,

I pre-trained two different models.(both of them output a vector with shape (1,4)). Additionally, I have another  external vector with fixed values, with shape (1,2)
Now I want to use them together as follows,

1. Turn the two outputs from the pre trained dnns to matrix(didn't succeed with 'concat')- now the shape of the matrix will be (2,4).
2. Multiple the external vec with the matrix  - output shape now is supposed to be (1,4)
3. Train all this together.

I tried to merge the two models outputs with concatenate, then another merge layer with dot product between the first merge and the external vec. It doesn't let me because the first merge does not turn into matrix.

I'll be happy for some advise.

**Comment by bernardohenz** [comment]
1) When merging using concat, you must choose the axis with ```concat_axis```
2) The ```mul``` mode of merge is a element-wise multiplication (take two Tensor with equal sizes and multiply element by element)
3) I always recommend to design the net as a Graph when using merge. See a simple example:

```python
input1 = Input(shape=(1, 128, 128))
input2 = Input(shape=(1, 128, 128))

conv1 = Convolution2D(1, 3, 3, border_mode='same')(input1)
conv1 = Convolution2D(1, 3, 3, border_mode='same')(conv1)
conv1 = Convolution2D(1, 3, 3, border_mode='same')(conv1)

conv2 = Convolution2D(1, 3, 3, border_mode='same')(input2)
conv2 = Convolution2D(1, 3, 3, border_mode='same')(conv2)
conv2 = Convolution2D(1, 3, 3, border_mode='same')(conv2)

merged = merge([conv1,conv2],mode='concat',concat_axis=1) #or 'concat' or 'mul'

model = Model(input=[input1,input2],output=merged )
```

**Comment by Shlomke** [comment]
I found an answer here :[( https://github.com/fchollet/keras/issues/3225)]
